### PR TITLE
Provide the 'challenge' function in the prelude.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by leonardoalt** [comment]
test failing

**Comment by chriseth** [comment]
Hopefully fixed now.

**Review by georgwiese — APPROVED** [review]
LGTM!

File: std/prover.asm [hunk 1]
```diff
@@ -20,7 +20,7 @@ enum Query {
 
 /// Constructs a challenge object.
 /// The arguments are the proof stage and the id of the challenge, in this order.
-let challenge: int, int -> expr = [];
+let challenge: int, int -> expr = constr |st, id| std::prelude::challenge(st, id);
```
[the remark this answers is not in the document]
Comment (chriseth) [reply std/prover.asm:23]: Yes, it would.

**Pull request merged**